gym-report-server/src/test/java/com/example/gymreportserver/service/TrainingReportServiceImplTest.java

```diff
@@ -36,21 +34,25 @@ class TrainingReportServiceImplTest {
     @BeforeEach
     void setUp() {
         trainingReportService = new TrainingReportServiceImpl(trainingReportRepository);
-        trainingReport = new TrainingReport(trainerUsername, "test", "test", true, Map.of(2023, Map.of("DECEMBER", 2)), ReportType.ADD);
+        Map<String, Integer> month = new HashMap<>();
+        month.put("DECEMBER", 2);
+        Map<Integer, Map<String, Integer>> years = new HashMap<>();
+        years.put(2023, month);
+        trainingReport = new TrainingReport(trainerUsername, "test", "test", true, years, ReportType.ADD);
     }
 
     @Test
     void postReport() {
-        given(trainingReportRepository.findByTrainerUsername(trainerUsername)).willReturn(Optional.empty());
+        given(trainingReportRepository.findByTrainerUsername(trainerUsername)).willReturn(Optional.of(trainingReport));
         trainingReportService.postReport(new ReportRequest(trainerUsername, "test", "test", true, LocalDate.now(), 2, ReportType.ADD), "transactionId");
         verify(trainingReportRepository).findByTrainerUsername(trainerUsername);
-        verify(trainingReportRepository).save(any());
+        verify(trainingReportRepository).save(trainingReport);
```

Comment:
what does this actually check? shouldn't we check task save is called with some training report with extended duration/new month/year ( since we have given that some report is already in DB for this trainer
Also it would be nice to check both scenarios - if there are already smth in DB for this trainer and if there are no. Also it would be nice to verify some additional cases if there are already smth in DB for this trainer - if a training happened in the same month, in the same year but different moth, and different month and year